src/test/java/com/parkhomovsky/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,246 @@
+            .setId(3L)
+            .setTitle("Kobzar")
+            .setAuthor("Shevchenko")
+            .setIsbn("1315616")
+            .setPrice(new BigDecimal("15"))
+            .setDescription("Good UA book")
+            .setCoverImage("https://URL")
+            .setCategoryIds(Set.of(1L));
+    private static final CreateBookRequestDto CREATE_BOOK_REQUEST_DTO =
+            new CreateBookRequestDto()
+            .setTitle("Dota 2: Pain and Casino")
+            .setAuthor("U.N. LoLovich")
+            .setIsbn("978-0-9767736-6-5")
+            .setPrice(new BigDecimal("25"))
+            .setDescription("Book about why you shoudn`t play Dota 2."
+                    + " And why you can`t still find a girl.")
+            .setCoverImage("https://resources/bookImages/Dota-Pain.png")
+            .setCategoryId(Set.of(2L));
+    private static final CreateBookRequestDto INVALID_CREATE_BOOK_REQUEST_DTO =
+            new CreateBookRequestDto()
+            .setAuthor("U.N. LoLovich")
+            .setIsbn("978-0-9767736-6-5")
+            .setPrice(new BigDecimal("25"))
+            .setDescription("Book about why you shoudn`t play Dota 2."
+                    + " And why you can`t still find a girl.")
+            .setCoverImage("resources/bookImages/Dota-Pain.png")
+            .setCategoryId(Set.of(2L));
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @Sql(
+            scripts = "classpath:db-scripts/categories/add-comedy-to-categories-table.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD
+    )
+    @Sql(
+            scripts = "classpath:db-scripts/clear-books_connections-tables.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @DisplayName("Creating book with valid book params. Ok Status and DTO Expected")
+    void create_withValidBook_returnCreatedBookDto() throws Exception {
+        MvcResult mvcResult = mockMvc.perform(post("/books")
+                        .content(objectMapper.writeValueAsString(CREATE_BOOK_REQUEST_DTO))
+                        .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        BookDto actual =
+                objectMapper.readValue(mvcResult.getResponse().getContentAsString(), BookDto.class);
+
+        assertTrue(EqualsBuilder.reflectionEquals(RESPONSE_CREATED_DOTA_BOOK_DTO, actual, "id"));
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @Sql(
+            scripts = "classpath:db-scripts/categories/add-fiction-to-categories-table.sql",
```

Comment:
I suggest you to add all of your scripts to constants also

Reply:
You forgot to replace by constant here